ui/pages/settings/security-tab/profile-sync-toggle/profile-sync-toggle.tsx

```diff
@@ -26,6 +27,19 @@ import {
   TextVariant,
 } from '../../../../helpers/constants/design-system';
 import Preloader from '../../../../components/ui/icon/preloader/preloader-icon.component';
+import { getUseExternalServices } from '../../../../selectors';
+
+export function useEffectProfileSyncBasicFunctionalitySetting() {
+  const basicFunctionality: boolean = useSelector(getUseExternalServices);
+  const { setIsProfileSyncingEnabled } = useSetIsProfileSyncingEnabled();
+
+  // Effect - we need to toggle profile syncing off when basic functionality is off
+  useEffect(() => {
+    if (basicFunctionality === false) {
+      setIsProfileSyncingEnabled(false);
+    }
+  }, [basicFunctionality, setIsProfileSyncingEnabled]);
+}
 
 const ProfileSyncToggle = () => {
```

Comment:
(Prith Note) this is a little annoying, we have basically 2 of the same toggles. 1 for Onboarding, and another here. I wish we could use the same component (but I understand the use-case/logic is different.)